src/integrationTest/sam.test.ts

```diff
@@ -0,0 +1,181 @@
+            }
+            documentUri = await openSamProject()
+        })
+
+        it('Generates a template with a proper runtime', async () => {
+            const fileContents = readFileSync(`${projectFolder}/testProject/template.yaml`).toString()
+            assert.ok(fileContents.includes(`Runtime: ${projectSDK}`))
+        })
+
+        it('Fails to create template when it already exists', async () => {
+            const initArguments: SamCliInitArgs = {
+                name: 'testProject',
+                location: projectFolder,
+                runtime: projectSDK as SamLambdaRuntime
+            }
+            console.log(initArguments.location)
+            const samCliContext = getSamCliContext()
+            await runSamCliInit(initArguments, samCliContext.invoker).catch((e: Error) => {
+                assert(e.message.includes('directory already exists'))
+            })
+        }).timeout(TIMEOUT)
+
+        it('Invokes the run codelens', async () => {
+            const [runCodeLens] = await getCodeLensesOrTimeout()
+            assert.ok(runCodeLens.command)
+            const command = runCodeLens.command!
+            assert.ok(command.arguments)
+            const runResult: any | undefined = await vscode.commands.executeCommand(
+                command.command,
+                ...command.arguments!
+            )
+            assert.ok(runResult)
+            // tslint:disable: no-unsafe-any
+            const datum = runResult!.datum
+            assert.strictEqual(datum.name, 'invokelocal')
+            assert.strictEqual(datum.value, 1)
+            assert.strictEqual(datum.unit, 'Count')
+
+            assert.ok(datum.metadata)
+            const metadata = datum.metadata!
+            assert.strictEqual(metadata.get('runtime'), projectSDK)
+            assert.strictEqual(metadata.get('debug'), 'false')
+        }).timeout(TIMEOUT)
+
+        it('Invokes the debug codelens', async () => {
+            const [, debugCodeLens] = await getCodeLensesOrTimeout()
+            assert.ok(debugCodeLens.command)
+            const command = debugCodeLens.command!
+            assert.ok(command.arguments)
+            const runResult: any | undefined = await vscode.commands.executeCommand(
+                command.command,
+                ...command.arguments!
+            )
+            assert.ok(runResult)
+            const datum = runResult!.datum
+            assert.strictEqual(datum.name, 'invokelocal')
+            assert.strictEqual(datum.value, 1)
+            assert.strictEqual(datum.unit, 'Count')
+
+            assert.ok(datum.metadata)
+            const metadata = datum.metadata!
+            assert.strictEqual(metadata.get('runtime'), projectSDK)
+            assert.strictEqual(metadata.get('debug'), 'true')
+            // This timeout is significantly longer, mostly to accommodate the long first time .net debugger
+        }).timeout(TIMEOUT * 2)
+
+        after(async () => {
```

Comment:
Move underneath the `before` statement. We usually stack all before/after statements at the top of a `describe` section.

Reply:
sure moved after